repository/kafka/operator/templates/statefulset.yaml

```diff
@@ -23,7 +23,7 @@ spec:
       containers:
         - name: k8skafka
           imagePullPolicy: Always
-          image: mesosphere/kafka:0.3.0-2.3.0
+          image: mesosphere/kafka:0.3.1-2.3.0
```

Comment:
This should totally be `0.4.0` :)

Reply:
this is the docker image. And in this version, only custom prinicpal jar was added. That's why a patch bump.

As for KUDO Kafka Operator, we will probably go `1.0.0` for GA reasons. And also bump docker images.